src/handlers/psp.ts

```diff
@@ -213,121 +236,80 @@ const executeApiCalls = async (state: State): Promise<State> => {
   return { ...state, apiValuesBySubscriptionId };
 };
 
-const initializeProviders = async (state: State): Promise<State> => {
-  const { config, baseLogOptions } = state;
-
-  const airnodeWallet = ethers.Wallet.fromMnemonic(config.nodeSettings.airnodeWalletMnemonic);
-
-  const evmChains = config.chains.filter((chain) => chain.type === 'evm');
-  if (isEmpty(evmChains)) {
-    throw new Error('One or more evm compatible chains must be defined in the provided config');
+export const spawn = ({
+  providerSponsorSubscription,
+  baseLogOptions,
+  type,
+  stage,
+}: {
+  providerSponsorSubscription: ProviderSponsorSubscriptions;
+  baseLogOptions: utils.LogOptions;
+  type: 'local' | 'aws' | 'gcp';
+  stage: string;
+}) => {
+  // lambda.invoke is synchronous so we need to wrap this in a promise
+  switch (type) {
+    case 'local':
+      return new Promise((resolve, reject) => {
+        processSubscriptionsHandler({ providerSponsorSubscription, baseLogOptions }).then((data) => {
+          if (!data.ok) {
+            reject(data.message);
+          }
+          resolve(data.message);
+        });
+      });
+    case 'aws':
+      return new Promise((resolve, reject) => {
+        // Uses the current region by default
+        const lambda = new AWS.Lambda();
+
+        // AWS doesn't allow uppercase letters in lambda function names
+        const resolvedName = `airkeeper-${stage}-process-subscriptions`;
+
+        const options = {
+          FunctionName: resolvedName,
+          Payload: JSON.stringify({ providerSponsorSubscription, baseLogOptions }),
+        };
+        lambda.invoke(options, (err, data) => {
+          // Reject invoke and (unhandled) handler errors
+          if (err || data.FunctionError) {
+            reject(err || data.FunctionError);
+            return;
+          }
+
+          const parsedData: AWSHandlerResponse = JSON.parse(data.Payload as string);
+
+          // Reject non-ok results
+          if (!parsedData.ok) {
+            reject(parsedData.message);
+            return;
+          }
+
+          resolve(parsedData.message);
+        });
+      });
```

Comment:
I refactored the `spawn` function to return information from each `process-subscription` handler and resolve/reject accordingly.